### PR TITLE
[GraphBolt] add example of heterograph node classification

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/graphbolt/rgcn/hetero_rgcn.py [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+    # Sample neighbors for each seed node in the mini-batch.
+    datapipe = datapipe.sample_neighbor(graph, fanouts=fanouts)
+
+    # Fetch the features for each node in the mini-batch.
+    datapipe = datapipe.fetch_feature(features, {"paper": ["feat", "year"]})
+
+    # Move the mini-batch to the appropriate device.
+    datapipe = datapipe.copy_to(device)
+
+    return gb.MultiProcessDataLoader(datapipe, num_workers=num_workers)
```
Comment (peizhou001) [inline examples/graphbolt/rgcn/hetero_rgcn.py:100]: Will it not be stuck without `drop_last`?

File: examples/graphbolt/rgcn/hetero_rgcn.py [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+    datapipe = gb.ItemSampler(item_set, batch_size=batch_size, shuffle=shuffle)
+
+    # Sample neighbors for each seed node in the mini-batch.
+    datapipe = datapipe.sample_neighbor(graph, fanouts=fanouts)
+
+    # Fetch the features for each node in the mini-batch.
+    datapipe = datapipe.fetch_feature(features, {"paper": ["feat", "year"]})
+
+    # Move the mini-batch to the appropriate device.
+    datapipe = datapipe.copy_to(device)
```
Comment (frozenbugs) [inline examples/graphbolt/rgcn/hetero_rgcn.py:98]: Does this actually work?

File: examples/graphbolt/rgcn/hetero_rgcn.py [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+
+    return graph, features, train_set, valid_set, test_set, num_classes
+
+
+def create_dataloader(
+    graph, features, item_set, device, batch_size, fanouts, shuffle, num_workers
+):
+    """Create a GraphBolt dataloader for training, validation or testing."""
+
+    # Initialize the ItemSampler to sample mini-batches from the dataset.
```
Comment (frozenbugs) [inline examples/graphbolt/rgcn/hetero_rgcn.py:88]: Highlight how graphbolt handles hetero graph.

File: examples/graphbolt/rgcn/hetero_rgcn.py (hunk 2 above)
Comment (Rhett-Ying) [reply examples/graphbolt/rgcn/hetero_rgcn.py:98]: gpu not verified yet. But this is supposed to be canonical UX, so I leave it here.

File: examples/graphbolt/rgcn/hetero_rgcn.py (hunk 1 above)
Comment (Rhett-Ying) [reply examples/graphbolt/rgcn/hetero_rgcn.py:100]: drop_last is always set as `False` for now. very limited numbers of options are supported for now.

File: examples/graphbolt/rgcn/hetero_rgcn.py (hunk 3 above)
Comment (Rhett-Ying) [reply examples/graphbolt/rgcn/hetero_rgcn.py:88]: For `Itemsampler`, it's totally same as homograph from user's perspective. As for other parts that need special attention, I will add

**Comment by Rhett-Ying** [comment]
@frozenbugs @peizhou001 I prefer to make this example runnable first even with limited options. we could add more options step by step such as `drop_last`, `gpu` and so on.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**